cmd/argoexec/commands/artifact/delete.go

```diff
@@ -52,63 +53,121 @@ func NewArtifactDeleteCommand() *cobra.Command {
 	}
 }
 
-func deleteArtifacts(labelSelector string, ctx context.Context, artifactGCTaskInterface wfv1alpha1.WorkflowArtifactGCTaskInterface) error {
+type request struct {
+	Task             *v1alpha1.WorkflowArtifactGCTask
+	NodeName         string
+	ArtifactNodeSpec *v1alpha1.ArtifactNodeSpec
+}
+
+type response struct {
+	Task     *v1alpha1.WorkflowArtifactGCTask
+	NodeName string
+	Results  map[string]v1alpha1.ArtifactResult
+	Err      error
+}
 
+func deleteArtifacts(labelSelector string, ctx context.Context, artifactGCTaskInterface wfv1alpha1.WorkflowArtifactGCTaskInterface) error {
 	taskList, err := artifactGCTaskInterface.List(context.Background(), metav1.ListOptions{LabelSelector: labelSelector})
 	if err != nil {
 		return err
 	}
+	taskWorkers := env.LookupEnvIntOr(common.EnvExecGCWorkers, 4)
+
+	totalTasks := 0
```

Comment:
I wonder if a different term than "task" could be used, since it's a little confusing with the existing use of `task` to mean something different in the code.

Reply:
Done